examples/saas-clients/amazons3/working-with-amazons3-client/README.md

```diff
@@ -0,0 +1,396 @@
+                respondAndHandleError(caller, backendResponse, RESPOND_ERROR_MSG);
+            }
+        } else {
+            // Send the error response.
+            createAndSendErrorResponse(caller, <string>amazonS3Client.detail().message, CLIENT_CREATION_ERROR_MSG);
+        }
+    }
+}
+
+// Function to extract the object content from request payload
+function extractRequestContent(http:Request request) returns string|xml|json|byte[]|error {
+    string contentTypeStr = request.getContentType();
+    if (contentTypeStr.equalsIgnoreCase("application/json")) {
+        var jsonObjectContent = request.getJsonPayload();
+        if (jsonObjectContent is json) {
+            return jsonObjectContent;
+        }
+    }
+    if (contentTypeStr.equalsIgnoreCase("application/xml")) {
+        var xmlObjectContent = request.getXmlPayload();
+        if (xmlObjectContent is xml) {
+            return xmlObjectContent;
+        }
+    }
+    if (contentTypeStr.equalsIgnoreCase("text/plain")) {
+        var textObjectContent = request.getTextPayload();
+        if (textObjectContent is string) {
+            return textObjectContent;
+        }
+    }
+    var binaryObjectContent = request.getBinaryPayload();
+    if (binaryObjectContent is byte[]) {
+        return binaryObjectContent;
+    } else {
+        error err = error(ERROR_CODE, { message : INVALID_PAYLOAD_MSG });
+        return err;
+    }
+}
+
+// Function to create the error response.
+function createAndSendErrorResponse(http:Caller caller, string errorMessage, string respondErrorMsg) {
+    http:Response response = new;
+    //Set 500 status code.
+    response.statusCode = 500;
+    //Set the error message to the error response payload.
+    response.setPayload(<string> errorMessage);
+    respondAndHandleError(caller, response, respondErrorMsg);
+}
+
+// Function to send the response back to the client and handle the error.
+function respondAndHandleError(http:Caller caller, http:Response response, string respondErrorMsg) {
+    // Send response to the caller.
+    var respond = caller->respond(response);
+    if (respond is error) {
+        log:printError(respondErrorMsg, err = respond);
+    }
+}
+```
+### Deployment
+
+#### Deploying locally
+You can deploy the services that you developed above in your local environment. You can create the Ballerina executable archives (.balx) first as follows.
+
+**Building**
+
+Navigate to working-with-amazons3-client and execute the following command.
+```bash
+$ ballerina build working-with-amazons3-client
+```
+
+After the build is successful, there will be a message_transformation_service.balx file inside the target directory. You can execute it as follows.
+
+```bash
+$ ballerina run working-with-amazons3-client.balx
+```
+
+### Testing
+
+- Navigate to `working-with-amazons3-client`, and execute the following command to start the service:
+
+```ballerina
+   $ ballerina run working-with-amazons3-client.bal
```

Comment:
This command should be corrected as "ballerina run working_with_amazons3_client.bal"